api/environments/environment_handler.go

```diff
@@ -138,7 +140,7 @@ func (eh EnvironmentHandler) DeleteEnvironment(appName, envName string) error {
 	}
 
 	// idempotent removal of RadixEnvironment
-	err = eh.radixclient.RadixV1().RadixEnvironments().Delete(uniqueName, &metav1.DeleteOptions{})
+	err = eh.getServiceAccount().RadixClient.RadixV1().RadixEnvironments().Delete(uniqueName, &metav1.DeleteOptions{})
 	if err != nil && !errors.IsNotFound(err) {
```

Comment:
Also check if `err` is not nil and something else (i.e. failure on deleting due to whatever reason).